test/units/modules/test_iptables.py

```diff
@@ -1,1174 +1,1402 @@
+    )
+    mocker.patch(
+        "ansible.module_utils.basic.AnsibleModule.fail_json",
+        side_effect=fail_json,
+    )
+    jump_err_msg = "jump is TEE but all of the following are missing: gateway"
+    with pytest.raises(AnsibleFailJson, match=jump_err_msg) as exc:
+        iptables.main()
+    assert exc.value.args[0]["failed"]
+
+
+def test_jump_tee_gateway(mocker, mock_basic_commands):
+    """Using gateway when JUMP is set to TEE."""
+    set_module_args(
+        {
+            "table": "mangle",
+            "chain": "PREROUTING",
+            "in_interface": "eth0",
+            "protocol": "udp",
+            "match": "state",
+            "jump": "TEE",
+            "ctstate": ["NEW"],
+            "destination_port": "9521",
+            "gateway": "192.168.10.1",
+            "destination": "127.0.0.1",
+        }
+    )
+    commands_results = [
+        (0, "", ""),
+    ]
+    run_command = mocker.patch(
+        "ansible.module_utils.basic.AnsibleModule.run_command",
+        side_effect=commands_results,
+    )
+
+    with pytest.raises(SystemExit):
+        iptables.main()
+
+    assert run_command.call_count == 1
+    first_cmd_args_list = run_command.call_args_list[0]
+    assert first_cmd_args_list[0][0] == [
+        IPTABLES_CMD,
+        "-t",
+        "mangle",
+        "-C",
+        "PREROUTING",
+        "-p",
+        "udp",
+        "-d",
+        "127.0.0.1",
+        "-m",
+        "state",
+        "-j",
+        "TEE",
+        "--gateway",
+        "192.168.10.1",
+        "-i",
+        "eth0",
+        "--destination-port",
+        "9521",
+        "--state",
+        "NEW",
+    ]
+
+
+@pytest.mark.parametrize(
+    ("test_input"),
+    [
+        pytest.param('flags=ALL flags_set="ACK,RST,SYN,FIN"', id="tcp_flags_str"),
+        pytest.param({"flags": "ALL", "flags_set": "ACK,RST,SYN,FIN"}, id="tcp_flags_dict"),
+        pytest.param({"flags": ["ALL"], "flags_set": ["ACK", "RST", "SYN", "FIN"]}, id="tcp_flags_list"),
+    ],
+)
+def test_tcp_flags(mocker, mock_basic_commands, test_input):
+    """Test various ways of inputting tcp_flags."""
+    rule_data = {
+        "chain": "OUTPUT",
+        "protocol": "tcp",
+        "jump": "DROP",
+        "tcp_flags": test_input,
+    }
+
+    set_module_args(rule_data)
+
+    commands_results = [
+        (0, "", ""),
+    ]
+
+    run_command = mocker.patch(
+        "ansible.module_utils.basic.AnsibleModule.run_command",
+        side_effect=commands_results,
+    )
+
+    with pytest.raises(SystemExit):
+        iptables.main()
+
+    assert run_command.call_count == 1
+    first_cmd_args_list = run_command.call_args_list[0]
+    assert first_cmd_args_list[0][0] == [
+        IPTABLES_CMD,
+        "-t",
+        "filter",
+        "-C",
+        "OUTPUT",
+        "-p",
+        "tcp",
+        "--tcp-flags",
+        "ALL",
+        "ACK,RST,SYN,FIN",
+        "-j",
+        "DROP",
+    ]
+
+
+@pytest.mark.parametrize(
+    ("log_level"),
```

Comment:
no need to wrap a single string with parens
```suggestion
    "log_level",
```